src/lib/Transpose.hs

```diff
@@ -286,6 +287,8 @@ transposeAtom atom ct = case atom of
   TabLam _        -> notTangent
   DictCon _       -> notTangent
   DictTy _        -> notTangent
+  HandlerDictCon _ -> notTangent  -- TODO(alex): check correctness
+  HandlerDictTy _ -> notTangent   -- TODO(alex): check correctness
```

Comment:
Yeah this seems reasonable. AD should run on simplified IR, which I imagine should not have any user-defined effects anymore

Reply:
Indeed -- however, these are removed by #1071, which I merged into this branch